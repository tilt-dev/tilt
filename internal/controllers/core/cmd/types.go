package cmd

import "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"

type Cmd = v1alpha1.Cmd
type CmdSpec = v1alpha1.CmdSpec
type CmdStatus = v1alpha1.CmdStatus
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdStateTerminated = v1alpha1.CmdStateTerminated

const LabelManifest = v1alpha1.LabelManifest
