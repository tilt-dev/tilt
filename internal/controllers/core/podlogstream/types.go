package podlogstream

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PodLogStream = v1alpha1.PodLogStream
type PodLogStreamSpec = v1alpha1.PodLogStreamSpec
type PodLogStreamStatus = v1alpha1.PodLogStreamStatus
type ContainerLogStreamStatus = v1alpha1.ContainerLogStreamStatus
type ObjectMeta = metav1.ObjectMeta
