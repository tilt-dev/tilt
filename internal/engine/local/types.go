package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type ObjectMeta = metav1.ObjectMeta
