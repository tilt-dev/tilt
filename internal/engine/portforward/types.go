package portforward

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type PortForward = v1alpha1.PortForward
type PortForwardSpec = v1alpha1.PortForwardSpec
type PortForwardStatus = v1alpha1.PortForwardStatus
type ObjectMeta = metav1.ObjectMeta
type Forward = v1alpha1.Forward
