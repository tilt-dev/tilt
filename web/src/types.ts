import { Alert } from "./alerts"
import { Facet } from "./facets"

export enum SocketState {
  Loading,
  Reconnecting,
  Closed,
  Active,
}

export enum ResourceView {
  Log,
  Alerts,
  Facets = 2,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManualAfterInitial,
  TriggerModeManualIncludingInitial,
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
  NotApplicable = "not_applicable",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  Building, // Tilt is actively doing something (e.g., docker build or kubectl apply)
  Pending, // not building, healthy, or unhealthy, but presumably on its way to one of those (e.g., queued to build, or ContainerCreating)
  Healthy, // e.g., build succeeded and pod is running and healthy
  Unhealthy, // e.g., last build failed, or CrashLoopBackOff
  Warning, // e.g., an undismissed restart
  None, // e.g., a manual build that has never executed
}

export type SnapshotHighlight = {
  beginningLogID: string
  endingLogID: string
  text: string
}

export enum ShowFatalErrorModal {
  Default,
  Show,
  Hide,
}

export type Snapshot = {
  view: Proto.webviewView
  isSidebarClosed: boolean
  path?: string
  snapshotHighlight?: SnapshotHighlight | null
}

// A plaintext representation of a line of the log,
// with metadata to render it in isolation.
//
// The metadata should be stored as primitive fields
// so that React's default caching behavior will kick in.
export type LogLine = {
  // We assume that 'text' does not contain a newline
  text: string
  manifestName: string
  level: string
}
