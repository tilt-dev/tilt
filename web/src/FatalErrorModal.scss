@import "constants";
@import "ReactModal";

.FatalErrorModal-pane {
  display: flex;
  flex-direction: column;
  font-size: $font-size;
  padding-top: $spacing-unit * 0.75;
  padding-bottom: $spacing-unit;
  padding-left: $spacing-unit;
  padding-right: $spacing-unit;
  max-height: 80vh;
  overflow-y: auto;
}

.FatalErrorModal-title {
  margin: 0;
  font-family: $font-sans-serif;
  font-size: $font-size-small;
  text-transform: uppercase;
  background-color: $color-red;
  color: $color-white;
  padding-top: calc($spacing-unit / 4);
  padding-bottom: calc($spacing-unit / 4);
  padding-left: $spacing-unit;
  padding-right: $spacing-unit;
}
