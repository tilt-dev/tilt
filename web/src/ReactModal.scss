@import "constants";

.ReactModal__Overlay {
  z-index: $z-modal;
  display: flex;
  align-items: center;
  justify-content: center;
}