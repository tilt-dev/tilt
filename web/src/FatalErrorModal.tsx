import React, { PureComponent } from "react"
import Modal from "react-modal"
import "./FatalErrorModal.scss"
import { ShowFatalErrorModal } from "./types"

type props = {
  error: string | null | undefined
  showFatalErrorModal: ShowFatalErrorModal
  handleClose: () => void
}

export default class FatalErrorModal extends PureComponent<props> {
  render() {
    let showModal =
      Boolean(this.props.error) &&
      (this.props.showFatalErrorModal === ShowFatalErrorModal.Default ||
        this.props.showFatalErrorModal === ShowFatalErrorModal.Show)
    return (
      <Modal
        isOpen={showModal}
        className="FatalErrorModal"
        onRequestClose={this.props.handleClose}
      >
        <h2 className="FatalErrorModal-title">Fatal Error</h2>
        <div className="FatalErrorModal-pane">
          <p>Tilt has encountered a fatal error: {this.props.error}</p>
          <p>
            Once you fix this issue you'll need to restart Tilt. In the meantime
            feel free to browse through the UI.
          </p>
        </div>
      </Modal>
    )
  }
}
