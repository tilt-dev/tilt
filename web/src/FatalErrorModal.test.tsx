import React from "react"
import ReactDOM from "react-dom"
import ReactModal from "react-modal"
import renderer from "react-test-renderer"
import FatalErrorModal from "./FatalErrorModal"
import { ShowFatalErrorModal } from "./types"

const fakeHandleCloseModal = () => {}
let originalCreatePortal = ReactDOM.createPortal

describe("FatalErrorModal", () => {
  beforeEach(() => {
    // Note: `body` is used as the app element _only_ in a test env
    // since the app root element isn't available; in prod, it should
    // be set as the app root so that accessibility features are set correctly
    ReactModal.setAppElement(document.body)
    let mock: any = (node: any) => node
    ReactDOM.createPortal = mock
  })

  afterEach(() => {
    ReactDOM.createPortal = originalCreatePortal
  })

  it("doesn't render if there's no fatal error and the modal hasn't been closed", () => {
    const tree = renderer
      .create(
        <FatalErrorModal
          error={null}
          showFatalErrorModal={ShowFatalErrorModal.Default}
          handleClose={fakeHandleCloseModal}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does render if there is a fatal error and the modal hasn't been closed", () => {
    const tree = renderer
      .create(
        <FatalErrorModal
          error={"i'm an error"}
          showFatalErrorModal={ShowFatalErrorModal.Default}
          handleClose={fakeHandleCloseModal}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("doesn't render if there is a fatal error and the modal has been closed", () => {
    const tree = renderer
      .create(
        <FatalErrorModal
          error={"i'm an error"}
          showFatalErrorModal={ShowFatalErrorModal.Hide}
          handleClose={fakeHandleCloseModal}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
