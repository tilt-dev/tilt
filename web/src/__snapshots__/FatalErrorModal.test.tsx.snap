// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatalErrorModal does render if there is a fatal error and the modal hasn't been closed 1`] = `
<div
  className="ReactModal__Overlay ReactModal__Overlay--after-open"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    aria-modal={true}
    className="ReactModal__Content ReactModal__Content--after-open FatalErrorModal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="FatalErrorModal-title"
    >
      Fatal Error
    </h2>
    <div
      className="FatalErrorModal-pane"
    >
      <p>
        Tilt has encountered a fatal error: 
        i'm an error
      </p>
      <p>
        Once you fix this issue you'll need to restart Tilt. In the meantime feel free to browse through the UI.
      </p>
    </div>
  </div>
</div>
`;

exports[`FatalErrorModal doesn't render if there is a fatal error and the modal has been closed 1`] = `null`;

exports[`FatalErrorModal doesn't render if there's no fatal error and the modal hasn't been closed 1`] = `null`;
